modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java

```diff
@@ -1011,8 +1014,23 @@ public CommerceOrder reorderCommerceOrder(
 				commerceOrder.getCommerceOrderId(), QueryUtil.ALL_POS,
 				QueryUtil.ALL_POS);
 
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.getCommerceChannelByGroupId(
+				commerceOrder.getGroupId());
+
 		for (CommerceOrderItem commerceOrderItem : commerceOrderItems) {
-			if (commerceOrderItem.getParentCommerceOrderItemId() != 0) {
+			CPDefinition cpDefinition = commerceOrderItem.getCPDefinition();
+
+			CommerceChannelRel commerceChannelRel =
+				_commerceChannelRelService.fetchCommerceChannelRel(
+					CPDefinition.class.getName(),
+					cpDefinition.getCPDefinitionId(),
+					commerceChannel.getCommerceChannelId());
+
+			if ((cpDefinition.isChannelFilterEnabled() &&
+				 (commerceChannelRel == null)) ||
+				(commerceOrderItem.getParentCommerceOrderItemId() != 0)) {
+
```

Comment:
@victorcdp , you do not fetch commerceChannelRel until you first check isChannelFilterEnabled is true. Please revise and resend.